pmlc/rt/runtime_registry.h

```diff
@@ -2,80 +2,36 @@
 
 #pragma once
 
-#include <functional>
 #include <memory>
-#include <string>
-#include <unordered_map>
 #include <utility>
-#include <vector>
 
 #include "llvm/ADT/StringRef.h"
 
 #include "pmlc/rt/runtime.h"
 
 namespace pmlc::rt {
 
-// Factory is used to instantiate a Runtime.
-//
-// Factory is invoked on-demand (typically well after process initialization) to
-// instantiate a runtime, either as a specific request by the PlaidML caller or
-// for general device enumeration.  The registry is responsible for arranging
-// runtimes to be instantiated exactly once; Factory implementations do not need
-// to memoize their results.
-using Factory = std::function<std::shared_ptr<Runtime>()>;
-
-// Loaders instantiate Factories.
-//
-// PlaidML Loaders are (typically) simple, stateless functions, automatically
-// registered via LoaderRegistration when their containing library is loaded
-// into the process.
-//
-// Loaders are invoked on-demand (typically well after process initialization)
-// to determine the runtime factories they're able to provide.
-using Loader = std::function<std::unordered_map<std::string, Factory>()>;
-
 namespace details {
 
-// Creates a global registration for a loader.
-void registerLoader(llvm::StringRef id, Loader loader);
+void registerRuntime(llvm::StringRef id, std::unique_ptr<Runtime> runtime);
```

Comment:
Why is this a 'detail', why not just have a public API here.

Reply:
Done

Reply:
... it's a little dangerous to use, though.  If `plaidml::exec::init()` might have been called, the new runtime will not have its `init()` method called by the system, so the caller needs to be very careful.

Reply:
I'm going to make it a requirement that we make this function public, because in other contexts, I will want to call this directly without using static initializers.

Reply:
It remains Done.  :-)